pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -535,3 +536,13 @@ def teardown(self) -> None:
             self.lightning_module.cpu()
             # clean up memory
             torch.cuda.empty_cache()
+
+    def load_checkpoint(self, checkpoint_path: _PATH) -> Dict[str, Any]:
+        rank_zero_info(f"DistributedDataParallel has {self.num_processes} processes. Serializing to avoid CPU OOMs.")
+        for current_worker in range(self.num_processes):
+            if self.local_rank == current_worker and torch.distributed.is_initialized():
```

Comment:
```suggestion
            if self.local_rank == current_worker:
```
we ought to fail loudly if torch distributed isn't initialized. otherwise nothing will be loaded